### DISCUSSION TITLE
Deleting a glossary triggers the checks

### DISCUSSION THREAD

**Discussion by tomkolp** [WeblateOrg/weblate#5693] [General] [answered]
Deleting a component marked as glossary triggers checks. With 1000 components, it takes about 2-3 hours using all workers (24xcpu 100%). Is this behavior correct?

**Comment by nijel** [comment, marked as answer]
It is expected if you had enabled translation propagation.

The performance should be much better with 298547610de1cff4e34605b42494fbc454bb77c0.

**Reply by tomkolp** [reply]
I tested this patch, I noticed a 3-fold increase in check speed. It takes about an hour now. The load for an hour is very heavy, the dasboard takes about 30 seconds to load. The ideal solution would be to be able to schedule the removal of the component during the night.

**Reply by nijel** [reply]
You can trigger the removal anytime it suits you ;-). I don't think Weblate should do any scheduling here - users will expect the checks to match reality, so they should be updated ASAP.

If you are worried about server load, decrease the number of Celery workers. It will take longer to complete, but it will not consume all CPU cures.

**Reply by tomkolp** [reply]
Will it speed up the loading of the dasbord if I reduce the number of workers?

**Reply by nijel** [reply]
Most likely yes, but it depends on what is the actual bottleneck in that case.
